### ISSUE TITLE
Discuss further for shard health states on INSERT INTO ... SELECT queries

### ISSUE THREAD

**Issue by onderkalaci** [citusdata/citus#954]
We currently error out if any of the placements on the `SELECT` query is invalid. 

Why we need such a restriction: If we `SELECT` from an `unhealthy` placement and `INSERT` that into a `healthy` placement, the inserted placement becomes `unhealthy` since it got data from a stale/old placements.

Note that if `replication factor = 1`, we never have placements with `unhealthy` states. So, this issue applies only for `replication factor > 1`.

An example to illustrate the current behavior:

```SQL

ALTER SEQUENCE pg_catalog.pg_dist_shardid_seq RESTART 1000000;

CREATE TABLE test_1
(
    key int,
    value text
);


CREATE TABLE test_2
(
    key int,
    value text
);

SELECT create_distributed_table('test_1', 'key');
SELECT create_distributed_table('test_2', 'key');

-- this works fine
INSERT INTO test_2 SELECT * FROM test_1;

-- now, change one of the shardstate of `test_1`
UPDATE pg_dist_shard_placement SET shardstate = 3 WHERE shardid = 1000000 AND nodeport = 9700;

-- this fails since one of the placements of `test_1` has invalid status
INSERT INTO test_2 SELECT * FROM test_1;
ERROR:  cannot perform distributed planning for the given modification
DETAIL:  Insert query cannot be executed on all placements for shard 1000032

```

**Alternative:**
One another option would be not to error out and mark the placements that are INSERTed into as `unhealthy`. For example, mark the placement with shardid `1000032` and nodeport 9700 `unhealthy` in the above example.

Do we have any other options here?

**Comment by onderkalaci** [comment]
Just a note I've realized: 
```DETAIL:  Insert query cannot be executed on all placements for shard 1000032``` should be 
```DETAIL:  Insert query could not be executed on all placements for shard 1000032``` 
with respect to the postgres [Error Message Style Guide](https://www.postgresql.org/docs/9.5/static/error-style-guide.html
)

**Comment by marcocitus** [comment]
No longer relevant after #5381
